src/org/toml/lang/icons/TomlIconProvider.kt

```diff
@@ -0,0 +1,25 @@
+package org.toml.lang.icons
+
+import com.intellij.ide.IconProvider
+import com.intellij.psi.PsiElement
+import org.toml.lang.core.psi.TomlFile
+import javax.swing.Icon
+
+
+public class TomlIconProvider : IconProvider() {
+    override fun getIcon(element: PsiElement, flags: Int): Icon? {
+        return when(element) {
+            is TomlFile -> getFileIcon(element)
+            else -> null
+        }
+    }
+
+    fun getFileIcon(element: TomlFile): Icon? {
```

Comment:
In `Kotlin` everything is public by default. Is this `fun` meant to be private? If it is, then use `private fun ...`